docs/DeveloperGuide.md

```diff
@@ -1,34 +1,890 @@
+    5. Lastly, a String called `feedbackToUser` will be returned to the user to inform the user of the outcome of the command. 
+4. The details will then be printed onto the console using `Ui#showResultToUser(result)`.
+
+
+##### Design Considerations
+Aspect: How is the `DeleteRecipeCommand` initialise. <br>
+<br>
+Alternative 1 (Current Choice): Usage of 2 constructors <br>
+
+|     |     |
+|-----|-----|
+|**Pros** | This gives us more flexibility on what object can be created with different variables since there are two methods of recipe deletion. |  
+|**Cons** | There is an overload of constructors.|
+
+Rationale for using this: As there are two different types of deletion, it would be simpler and increase cohesion as it is more easier to express these constructors' functionality at a higher level.
+
+Alternative 2: Usage of 1 constructor <br>
+
+|     |     |
+|-----|-----|
+|**Pros** |The Parser can call for one main default constructor. |
+|**Cons** | The single constructor will need to deal with 2 different methods of deletion, causing the constructor to have more than one purpose.|
+
+
+#### 4.2.5. Search for recipe based on keyword(s)
+
+The search for recipe feature allows the user to find recipes using a keyword in the recipe’s list.  
+For example, `searchrecipe Chicken` will find all recipes that contain `Chicken`.  
+
+
+##### Implementation
+
+![SearchRecipeCommand](images/searchRecipeCommand1.png)
+
+The following steps explained sequence diagram for `searchrecipe` command:  
+1. The user enters `searchrecipe Chicken`.  
+2. `KitchenHelper` calls `Parser#parseUserCommand()`.  
+3. `SearchRecipeCommand` object is created with the keyword passed in.  
+4. `KitchenHelper` calls it own method `executeCommand()` to execute the method in `SearchRecipeCommand#execute()`.  
+5. On `SearchRecipeCommand#execute()`, display the list of recipe's name that matches the keyword.
+
+##### Design considerations:
+
+Aspects: How `searchrecipe` executes:  
+
+- Alternative 1 (current choice): Find if the keyword is part of the substring of the recipe’s name 
+and returns the recipe’s name and the index of recipe in the recipe’s list.  
+
+|     |     |
+|-----|-----|
+|**Pros** | 1. Easy to find similar recipe by their name.|  
+|**Cons** | 1. Only shows the different recipe that contains the keyword.|
+
+- Alternative 2: Find the keyword within the recipe’s ingredient.  
+
+|     |     |
+|-----|-----|
+|**Pros** | 1. More accurate searching of the recipe that uses the ingredients.|  
+|**Cons** | 1. Could be more memory intensive to find if the list is huge.|
+
+#### 4.2.5. Cooking a recipe
+##### Implementation
+##### Design considerations
+Aspect: Preparing the deduction of ingredients when cooking a recipe
+Alternative  1 (current choice): Checks for existence of recipe, existence of ingredients for the specified recipe and sufficiency of ingredients
```

Comment:
Minor formatting error here as well. It does not show up on the website correctly.